src/walktree.js

```diff
@@ -1,49 +1,60 @@
-const fs = require('fs');
+var fs = require('fs');
+var path = require('path');
 
-const walkTree = function walkTree(opts) {
-    const path = opts.root;
-    const ignored = opts.ignore;
-    const create = opts.create;
-    const countFolders = opts.countFoldersAsFiles;
-    const fileName = opts.file;
 
-    walkTreeRecursive(path);
+function walkTree (opts) {
+    var create = opts.create;
+    var ignored = opts.ignore;
+    var countFolders = opts.countFoldersAsFiles;
+    var nameOfPlaceholder = opts.file;
 
-    function walkTreeRecursive(path) {
-        let elements = fs.readdirSync(path);
-        let fileCount = 0;
+    walkTreeRecursive(opts.root);
 
-        while (elements.length > 0) {
-            let element = elements.pop();
-            let elementPath = path + '/' + element;
-
-            // Skip ignored folder and tries next folder in line
-            if (ignored.includes(element))
-                continue;
+    function walkTreeRecursive(dirPath) {
+        fs.readdir(dirPath, function(err, elements) {
+            if (err) {
+              throw new Error('Failed to read the directory at: ' + dirPath);
+            }
 
-            let stat = fs.statSync(elementPath);
-            if (stat.isDirectory()) {
-                if (create && countFolders)
-                    fileCount++;
+            var fileCount = 0;
 
-                walkTreeRecursive(elementPath);
+            elements.forEach(function(element) {
+                var fullPathToElement = path.join(dirPath, element);
```

Comment:
Here's a good example of why I like `path.join()`. The old version used string concatenation using a plus sign, which is a very common pattern. But when using paths it can be troublesome. Case in point: if the user somehow uses `/path/to/my/dir/` as their root path instead of `/path/to/my/dir`, your concatenated path becomes `/path/to/my/dir//file`, which can produce unexpected behaviour!